osmoutils/accum/accum_test.go

```diff
@@ -318,3 +321,439 @@ func (suite *AccumTestSuite) TestClaimRewards() {
 		})
 	}
 }
+
+func (suite *AccumTestSuite) TestAddToPosition() {
+	type testcase struct {
+		startingNumShares sdk.Dec
+		startingUnclaimedRewards sdk.DecCoins
+		newShares   sdk.Dec
+
+		// accumInit and accumDelta specify the initial accum value 
+		// and how much it has changed since the position being added 
+		// to was created
+		accumInit	sdk.DecCoins
+		accumDelta  sdk.DecCoins
+
+		// Address does not exist
+		addrDNE bool
+		expPass bool
+	}
+
+	tests := map[string]testcase{
+		"zero shares with no new rewards": {
+			startingNumShares: sdk.ZeroDec(),
+			startingUnclaimedRewards: sdk.NewDecCoins(),
+			newShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(),
+			// unchanged accum value, so no unclaimed rewards
+			accumDelta: sdk.NewDecCoins(),
+			expPass: true,
+		},
+		"non-zero shares with no new rewards": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(),
+			newShares: sdk.NewDec(10),
+			accumInit: sdk.NewDecCoins(),
+			// unchanged accum value, so no unclaimed rewards
+			accumDelta: sdk.NewDecCoins(),
+			expPass: true,
+		},
+		"non-zero shares with new rewards in one denom": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(),
+			newShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(),
+			// unclaimed rewards since last update
+			accumDelta: sdk.NewDecCoins(initialCoinDenomOne),
+			expPass: true,
+		},
+		"non-zero shares with new rewards in two denoms": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(),
+			newShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(),
+			accumDelta: sdk.NewDecCoins(initialCoinDenomOne, initialCoinDenomTwo),
+			expPass: true,
+		},
+		"non-zero shares with both existing and new rewards": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(sdk.NewDecCoin(denomOne, sdk.NewInt(11)), sdk.NewDecCoin(denomTwo, sdk.NewInt(11))),
+			newShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(),
+			accumDelta: sdk.NewDecCoins(initialCoinDenomOne, initialCoinDenomTwo),
+			expPass: true,
+		},
+		"non-zero shares with both existing (one denom) and new rewards (two denoms)": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(initialCoinDenomOne),
+			newShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(),
+			accumDelta: sdk.NewDecCoins(initialCoinDenomOne, initialCoinDenomTwo),
+			expPass: true,
+		},
+		"non-zero shares with both existing (one denom) and new rewards (two new denoms)": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(initialCoinDenomOne),
+			newShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(),
+			accumDelta: sdk.NewDecCoins(initialCoinDenomTwo, sdk.NewDecCoin("baz", sdk.NewInt(10))),
```

Comment:
nit: we have `denomTwo` defined to be potentially used in place of "baz"

Reply:
I was using `denomOne` in place of "foo" and `denomTwo` in place of "bar", so "baz was actually a third denom. Will move it into a new `denomThree` variable